requirements.txt

```diff
@@ -0,0 +1,9 @@
+numpy
```

Comment:
I made a PR to support that use case: https://github.com/empa-scientific-it/python-tutorial/pull/23

Reply:
I would prefer to rely exclusively on venv and move all the requirements on requirements.txt. If this is fine by you, I will create a corresponding PR.

Reply:
Actually, I disagree that venv is a better alternative. I would very much prefer to use conda, cause it is so easy to work with it right out of the box. 

Here are some useful links to read: 

https://stackoverflow.com/questions/67065725/conda-env-vs-venv-pyenv-virtualenv-etc
The summary the person made reads as follows:

> In short, if you don't have a strong preference already, conda is more robust than venv or pip, can be combined with pip, and is probably the better default option. That said, if you already have a strong preference it means you likely already know how to do what you want, so it's unlikely to be worth it to change.

Also, the link that the answer has led to a nice [introduction to conda](https://whiteboxml.com/blog/the-definitive-guide-to-python-virtual-environments-with-conda), where one could also find:

> venv and conda differ in two main points:
>
> conda is more than a Python virtual environment manager. It is a generic virtual environment that supports much more than Python. Python installed in a conda environment is a real Python executable, not a link to your System Python (as is the case in other alternatives), so its version can be whatever you want, is not constrained by the System Python.
>
> venv is limited to installing packages using pip, while using conda you have both pip and conda package installer available.

Reply:
> Actually, I disagree that venv is a better alternative. I would very much prefer to use conda, cause it is so easy to work with it right out of the box.
> 
> Here are some useful links to read:
> 
> https://stackoverflow.com/questions/67065725/conda-env-vs-venv-pyenv-virtualenv-etc The summary the person made reads as follows:
> 
> > In short, if you don't have a strong preference already, conda is more robust than venv or pip, can be combined with pip, and is probably the better default option. That said, if you already have a strong preference it means you likely already know how to do what you want, so it's unlikely to be worth it to change.
> 
> Also, the link that the answer has led to a nice [introduction to conda](https://whiteboxml.com/blog/the-definitive-guide-to-python-virtual-environments-with-conda), where one could also find:
> 
> > venv and conda differ in two main points:
> > conda is more than a Python virtual environment manager. It is a generic virtual environment that supports much more than Python. Python installed in a conda environment is a real Python executable, not a link to your System Python (as is the case in other alternatives), so its version can be whatever you want, is not constrained by the System Python.
> > venv is limited to installing packages using pip, while using conda you have both pip and conda package installer available.

While you are right regarding conda being more than just a virtual environment, I don't think this is a really major point here as we are going to work on binder. 
My main complaint regarding conda is the slow speed in initialising an environment and solving dependencies. 

This can partially mitigated using mamba, but you need to install this via conda first.  I guess I will keep using venv locally, I don't mind if you prefer conda for the project on binder. Otherwise, the more robust approach for distribution would be to create a docker container with the required python binaries

Reply:
From my side, there is no issue with using pip inside a conda environment. I can modify the `environment.yml` file to use pip as the installer. This approach might work as a middle-ground solution that satisfies both our needs.

My problem with venv is that it is not fully isolated. As the document says - venv python is "a link to your System Python". I had issues with that in the past, I am not willing to come back.

Reply:
> From my side, there is no issue with using pip inside a conda environment. I can modify the `environment.yml` file to use pip as the installer. This approach might work as a middle-ground solution that satisfies both our needs.
> 
> My problem with venv is that it is not fully isolated. As the document says - venv python is "a link to your System Python". I had issues with that in the past, I am not willing to come back.
 That works well, let's stick with this solution.

Reply:
> That works well, let's stick with this solution.

Ok, cool 👍 . Updated #23 - please have a look. Only necessary (and unavoidable) packages are left for the coda installer - the rest goes through pip.